File: ABC01_99/ABC14/c.py
def main():
    N = int(input())
    r = [list(map(int, input().split()))for i in range(N)]
    c = [0] * 1100000
    for i in range(N):
        c[r[i][0]] += 1
        c[r[i][1] + 1] -= 1

    s = [0] * 1100000
    s[0] = c[0]
    for i in range(1000100):
        s[i] = s[i - 1] + c[i]
    ans = max(s)
    print(ans)


if __name__ == "__main__":
    main()

File: ABC01_99/ABC14/c.cpp
#include <iostream>
#include <vector>
using namespace std;
using ll = long long;
const int MAX_N = 1100000;

int main() {
  int N;
  cin >> N;
  vector<pair<ll, ll>> r(N);
  for (int i = 0; i < N; i++) {
    ll a, b;
    cin >> a >> b;
    r[i] = make_pair(a, b);
  }

  vector<ll> c(MAX_N, 0);
  for (int i = 0; i < N; i++) {
    c[r[i].first] += 1;
    c[r[i].second + 1] -= 1;
  }

  vector<ll> sum(MAX_N, 0);
  sum[0] = c[0];
  for (int i = 1; i < MAX_N; i++) {
    sum[i] = sum[i - 1] + c[i];
  }

  ll ans = 0;
  for (int i = 0; i <= MAX_N; i++) ans = max(ans, sum[i]);
  cout << ans << endl;
  return 0;
}